lib/tokens.js

```diff
@@ -449,6 +465,41 @@ export async function resolveToInternalId({token}) {
   return {internalId};
 }
 
+/**
+ * Updates a token batch based off of `internalId`.
+ *
+ * @param {object} options - Options to use.
+ * @param {string} options.internalId - The internal ID of the token batch.
+ * @param {number} options.minAssuranceForResolution - Minimum level of
+ *   assurance required for token resolution.
+ *
+ * @returns {Promise<boolean>} Returns true if update occurred.
+ */
+export async function updateBatches({internalId, minAssuranceForResolution}) {
+  const collection = database.collections['tokenization-tokenBatch'];
+  const query = {
+    'tokenBatch.internalId': internalId,
+    'tokenBatch.remainingTokenCount': {$gt: 0}
+  };
+  const records = await collection.find(query).toArray();
+  let result;
+  for(const record of records) {
+    const updateQuery = {
+      'tokenBatch.id': record.tokenBatch.id
+    };
+    const $set = {
+      'meta.updated': Date.now(),
+      'tokenBatch.minAssuranceForResolution': minAssuranceForResolution
+    };
+
+    result = await collection.updateOne(
+      updateQuery, {$set}, database.writeOptions);
+  }
```

Comment:
This isn't the right way to update multiple records. Instead of finding and retrieving them -- use `updateMany` and do a single call.